### ISSUE TITLE
installed_rpms missing EPOCH

### ISSUE THREAD

**Issue by ggainey** [RedHatInsights/insights-core#1431]
installed_rpms is generated with this query-format:

`"command": "/bin/rpm -qa --qf='%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}\t%{INSTALLTIME:date}\t%{BUILDTIME}\t%{VENDOR}\t%{BUILDHOST}\tDUMMY\t%{SIGPGP:pgpsig}\n'"`

'epoch' isn't always used - but when it is, it's important.

Can we change the -qf to

`"command": "/bin/rpm -qa --qf='%{NAME}-%{EPOCH}:%{VERSION}-%{RELEASE}.%{ARCH}\t%{INSTALLTIME:date}\t%{BUILDTIME}\t%{VENDOR}\t%{BUILDHOST}\tDUMMY\t%{SIGPGP:pgpsig}' | sed -e 's/(none)://' | sort\n'"`

instead? The sed-cmd is to handle the case when epoch is not specified at all for an RPM - so you don't get '-(none):' as an epoch.

**Comment by bfahr** [comment]
After investigating it appears that because the current parser already supports a JSON format for the rpm query we could just capture it using a query like this instead:
```
$ /bin/rpm -qa --qf='\{"name":"%{NAME}","epoch":"%{EPOCH}","version":"%{VERSION}","release":"%{RELEASE}","arch":"%{ARCH}","installtime":"%{INSTALLTIME:date}","buildtime":"%{BUILDTIME}","vendor":"%{VENDOR}","buildhost":"%{BUILDHOST}","sigpgp":"%{SIGPGP:pgpsig}"\}\n'
```
See PR #1484

**Comment by bfahr** [comment]
@ggainey the PR for this has been merged and it went into production yesterday.

**Comment by ggainey** [comment]
Insights-classic calls the vmaas-webservice with the NEVRA-list uploaded by the client. That call wants a NEVRA list in the format specified in this issue (i.e., NAME-EPOCH:VERSION-RELEASE.ARCH ). Right now, in production, Insights-classic is returning incorrect results, because it's missing EPOCH where it's required (i.e., non-zero epoch RPMs)

If Classic can take the data in the format the PR implemented, and send it to vmaas correctly, then we're good - I was just trying to make it easier on Classic.

**Comment by bfahr** [comment]
Reopening to include new methods for epoch in `installed_rpms`.

**Comment by xiangce** [comment]
Solved in #1498
